pkg/pubsub/adapter/converters/pubsub.go

```diff
@@ -58,5 +62,74 @@ func convertPubSub(ctx context.Context, msg *cepubsub.Message, sendMode ModeType
 			}
 		}
 	}
+	// If send mode is Push, convert to Pub/Sub Push payload style.
+	if sendMode == Push {
+		// set the content type to something that can be handled by codec.go
+		event.SetDataContentType("application/json")
+		event = convertToPush(ctx, event, msg.Attributes)
+	}
 	return &event, nil
 }
+
+// PushMessage represents the format Pub/Sub uses to push events.
+type PushMessage struct {
+	// Subscription is the subscription ID that received this Message.
+	Subscription string `json:"subscription"`
+	// Message holds the Pub/Sub message contents.
+	Message *PubSubMessage `json:"message,omitempty"`
+}
+
+// PubSubMessage matches the inner message format used by Push Subscriptions.
+type PubSubMessage struct {
+	// ID identifies this message. This ID is assigned by the server and is
+	// populated for Messages obtained from a subscription.
+	// This field is read-only.
+	ID string `json:"id,omitempty"`
+
+	// Data is the actual data in the message.
+	Data interface{} `json:"data,omitempty"`
+
+	// Attributes represents the key-value pairs the current message
+	// is labelled with.
+	Attributes map[string]string `json:"attributes,omitempty"`
+
+	// The time at which the message was published. This is populated by the
+	// server for Messages obtained from a subscription.
+	// This field is read-only.
+	PublishTime time.Time `json:"publish_time,omitempty"`
+}
+
+// convertToPush convert an event to a Pub/Sub style Push payload.
+func convertToPush(ctx context.Context, event cloudevents.Event, attrs map[string]string) cloudevents.Event {
+	logger := logging.FromContext(ctx).With(zap.Any("event.id", event.ID()))
+
+	tx := pubsubcontext.TransportContextFrom(ctx)
```

Comment:
some of this things are already done in the calling code... 
I think we should try to do all the common stuff at the beginning, such as creating the event with the spec version, id, time, etc... and then have an if the sendMode is push... you set up the content type and the proper event.Data with this push payload.... if it's not push, then we promote the msg.Attributes to extensions, if they are valid...
does it make sense?

Reply:
done. Note tests used to check the Push event also had the extensions, but probably not necessary to have them there.

Reply:
yeah, I think that was wrong... If we send push format, we shouldn't be promoting attributes to extensions also..